grpc/client/src/main/java/io/helidon/grpc/client/GrpcServiceClient.java

```diff
@@ -0,0 +1,338 @@
+
+import io.helidon.grpc.client.util.SingleValueResponseStreamObserverAdapter;
+
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.MethodDescriptor.MethodType;
+import io.grpc.stub.AbstractStub;
+import io.grpc.stub.ClientCalls;
+import io.grpc.stub.StreamObserver;
+
+/**
+ * A base class for gRPC Clients.
+ *
+ * @author Mahesh Kannan
+ */
+@ThreadSafe
+public class GrpcServiceClient {
+
+    private Channel channel;
+
+    private CallOptions callOptions;
+
+    private HashMap<String, GrpcMethodStub> methodStubs = new HashMap<>();
+
+    private ClientServiceDescriptor clientServiceDescriptor;
+
+    /**
+     * Builder to build an instance of {@link io.helidon.grpc.client.GrpcServiceClient}.
+     */
+    public static class Builder {
+
+        // This is the client being built.
+        private GrpcServiceClient grpcClient = new GrpcServiceClient();
+
+        /**
+         * Create a new instance that can be used to invoke methods on the service.
+         *
+         * @param channel                 the {@link Channel} to connect to the server
+         * @return This {@link }GrpcServiceClient.Builder} for fluent API.
+         */
+        public Builder channel(Channel channel) {
+            this.grpcClient.channel = channel;
+            return this;
+        }
+
+        /**
+         * Create a new instance that can be used to invoke methods on the service.
+         *
+         * @param callOptions             the {@link CallOptions}
+         * @return This {@link }GrpcServiceClient.Builder} for fluent API.
+         */
+        public Builder callOptions(CallOptions callOptions) {
+            this.grpcClient.callOptions = callOptions;
+            return this;
+        }
+
+        /**
+         * Create a new instance that can be used to invoke methods on the service.
+         *
+         * @param clientServiceDescriptor The {@link io.helidon.grpc.client.ClientServiceDescriptor} that describes the service.
+         * @return This {@link }GrpcServiceClient.Builder} for fluent API.
+         */
+        public Builder clientServiceDescriptor(ClientServiceDescriptor clientServiceDescriptor) {
+            this.grpcClient.clientServiceDescriptor = clientServiceDescriptor;
+            return this;
+        }
+
+        /**
+         * Build an instance of {@link io.helidon.grpc.client.GrpcServiceClient}.
+         * @return an instance of {@link io.helidon.grpc.client.GrpcServiceClient}.
+         */
+        public GrpcServiceClient build() {
+            return new GrpcServiceClient(grpcClient);
+        }
+    }
+
+    /**
+     * Creates a new {@link GrpcServiceClient.Builder}.
+     * @return A new instance of {@link GrpcServiceClient.Builder}.
+     */
+    public static GrpcServiceClient.Builder builder() {
+        return new GrpcServiceClient.Builder();
+    }
+
+    private GrpcServiceClient() { }
+
+    private GrpcServiceClient(GrpcServiceClient other) {
```

Comment:
This method needs a @SuppressWarnings("unchecked") annotation

Reply:
This method is removed because the builder pattern changed.